### PR TITLE
Fail build on linting error

### PR DESCRIPTION
**Pull request by SimenB**
With this change, the build actually fails.

### CODE REVIEW

**Comment by SimenB** [comment]
@lazd Rebased against master and fixed merge error :smile:

File: .jscsrc [hunk 1]
```diff
@@ -5,6 +5,7 @@
   },
   "disallowQuotedKeysInObjects": true,
   "disallowPaddingNewlinesInBlocks": true,
+  "disallowNewlineBeforeBlockStatements": true,
```
Comment (lazd) [inline .jscsrc]: I prefer:

```js
if (whatever) {
  // stuff
}
else {
  // stuff
}
```

Which I think this change forbids :P

Comment (lazd) [inline .jscsrc]: Hooray!

File: package.json [hunk 2]
```diff
@@ -16,14 +16,14 @@
...
+    "jshint": "^2.7.0",
     "mocha": "^2.2.4",
     "rimraf": "^2.3.2",
-    "should": "^5.2.0"
+    "should": "^6.0.1"
   },
   "scripts": {
     "clean": "rimraf coverage/",
-    "lint": "jscs index.js test/ & jshint index.js test/",
+    "lint": "jscs index.js test/ && jshint index.js test/",
```
Comment (SimenB) [inline package.json]: This extra `&` is the real change. Stupid forking not playing nice :laughing:

File: .jshintrc [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "node": true,
   "latedef": true,
-  "undef": true
+  "undef": true,
+  "unused": true
```
Comment (SimenB) [inline .jshintrc]: Discovered 2 missing, so just quickly added this :smile:

File: index.js [hunk 4]
```diff
@@ -6,7 +6,6 @@ var gutil = require('gulp-util');
 var c = gutil.colors;
 var error = gutil.PluginError;
 var es = require('event-stream');
-var fs = require('fs');
```
Comment (SimenB) [inline index.js]: This is the reason coverage is down

**Comment by SimenB** [comment]
@lazd I removed a line of code that was "tested", that's why coverage went down

**Comment by lazd** [comment]
@SimenB, I've been slacking, but let's get these rebased/merged and release this today.

**Comment by SimenB** [comment]
@lazd Awesome! Rebased now :smile:

**Comment by lazd** [comment]
So it looks like we should remove `failReporter` now. Note that `gulp-jshint` uses a `failReporter` mechanism to fail the build as well: https://github.com/spalger/gulp-jshint#fail-reporter

I wonder if we're deviating from an accepted convention with this change?

**Comment by lazd** [comment]
Ok, simple build only change. Merging!

**Pull request merged**